certora/specs/authorizations.spec

```diff
@@ -0,0 +1,263 @@
+using Utils as utils
+
+methods {
+    totalSupply() returns (uint256) envfree
+    balanceOf(address) returns (uint256) envfree
+    allowance(address, address) returns (uint256) envfree
+    owner() returns (address) envfree
+    getUserRoles(address) returns (bytes32) envfree
+    isCapabilityPublic(bytes4) returns (bool) envfree
+    getRolesWithCapability(bytes4) returns (bytes32) envfree
+    doesUserHaveRole(address, uint8) returns (bool) envfree
+    doesRoleHaveCapability(uint8, bytes4) returns (bool) envfree
+    utils.to_bytes4(uint32) returns (bytes4) envfree
+}
+
+
+// AUTHORIZATION STORAGE 
+
+// The authorization variables are: owner, isCapabilityPublic, getUserRoles and getRolesCapability.
+
+// owner variable
+
+rule ownerChanging() {
+    env e;
+    method f; calldataarg args; bytes4 newOwner;
```

Comment:
In this case I feel like it does not add much clarity, because the function `f` is arbitrary. The idea is that in a rule like
```
rule proveThingsAboutSetOwner(address newOwner)
```
we immediately see that we will be applying `setOwner` to `newOwner`

Reply:
Ok I see